lib/Dialect/FIRRTL/Transforms/Lint.cpp

```diff
@@ -0,0 +1,58 @@
+//===- Lint.cpp -------------------------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "PassDetails.h"
+#include "mlir/IR/Threading.h"
+#include "llvm/ADT/APSInt.h"
+
+using namespace mlir;
+using namespace circt;
+using namespace firrtl;
+
+namespace {
+struct LintPass : public LintBase<LintPass> {
+  void runOnOperation() override {
+    auto fModule = getOperation();
+    (void)failableParallelForEach(
+        fModule.getContext(), fModule.getOps(),
+        [&](Operation &op) { return checkAssert(&op); });
+  };
```

Comment:
Consider marking all analyses preserved to be a good citizen re:not needing to recompute cached analyses.